Assets/Project/Core/Actors/Obstacles/ObstacleSpawner.cs

```diff
@@ -0,0 +1,38 @@
+using System.Collections.Generic;
+using UnityEngine;
+
+public class ObstacleSpawner : MonoBehaviour, ISpawner
+{
+    //TODO: SpawingData must be configurable on a separate data class
+    [SerializeField] private GameObject obstacleToSpawn;
+    [SerializeField] private ObstacleData initialObstacleData;
+
+    [SerializeField] private readonly List<Obstacle> spawnedObstacles = new List<Obstacle>();
+    
+    public void StartSpawning()
+    {
+        CreateObstacle();
```

Comment:
Faz sentido ter um Object Pool, mas eu só to em dúvida das responsabilidades: será que não deveria haver apenas 1 Spawner ao invés de vários?

Reply:
Atualmente o spawner não controla o comportamento dos obstáculos, eles literalmente só spawnam. E eu não instancio vários obstáculos também. Cada vez que uma parede chega ao final da sua trajetória ela mesma se reposiciona e reinicia seu ciclo, os spawners só instanciam os objetos uma vez.

Ou seja, pra ter uma mecânica de várias paredes em velocidades diferentes e posições diferentes, eu configuro essas informações em cada spawner, e só tenho 1 parede pra cada spawner. A vantagem que eu vejo nisso é tirar do spawner a responsabilidade de reposicionar o objeto, já que por agora esse comportamento é só da "Wall". Outros obstáculos podem ser 1 shot, não se repetem, outros podem ter comportamentos exclusivos. Faz sentido?